build_scripts/windows/scripts/az

```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+AZ_INSTALLER=MSI "$(dirname "$(dirname "$0")")/python.exe" -IBm azure.cli "$@"
```

Comment:
Let's not use `$0`. There is an article talking about why: http://mywiki.wooledge.org/BashFAQ/028. Bash returns the actual command typed in by the user instead of the fully qualified path. If I am already in `build_scripts/windows/scripts/` and run `./az`, `$0` will be `./az`.

Also, `dirname` cannot go further than the current dir `.` - it will resolve to `./python.exe` which is not correct.

```
$ dirname .
.
```

Instead, let's follow this pattern and use `BASH_SOURCE` as 

https://github.com/Azure/azure-cli/blob/6354cce02b013bb7e25ab7fee8cba19124a087d3/scripts/install_full.sh#L15

Reply:
good suggestion.